### PR TITLE
WFLY-18377 Upgrade Infinispan to 14.0.14.Final

### PR DESCRIPTION
**Pull request by pferraro**
https://issues.redhat.com/browse/WFLY-18377

### CODE REVIEW

File: clustering/infinispan/client/spi/src/main/java/org/wildfly/clustering/infinispan/client/near/CaffeineNearCache.java [hunk 1]
```diff
@@ -43,18 +43,23 @@ public CaffeineNearCache(Cache<K, MetadataValue<V>> cache) {
...
-        this.map.putIfAbsent(key, value);
+    public boolean remove(K key) {
+        return this.map.remove(key) != null;
     }
 
     @Override
-    public boolean remove(K key) {
-        return this.map.remove(key) != null;
+    public boolean remove(K key, MetadataValue<V> value) {
+        return this.map.replace(key, value) != null;
```
Comment (jamezp) [inline clustering/infinispan/client/spi/src/main/java/org/wildfly/clustering/infinispan/client/near/CaffeineNearCache.java:57]: Should this be `this.map.remove(key ,value)`?

**Review by jamezp — CHANGES_REQUESTED** [review]

File: clustering/infinispan/client/spi/src/main/java/org/wildfly/clustering/infinispan/client/near/CaffeineNearCache.java (hunk 1 above)
Comment (pferraro) [reply clustering/infinispan/client/spi/src/main/java/org/wildfly/clustering/infinispan/client/near/CaffeineNearCache.java:57]: :facepalm: Another auto-completion victim...

Comment (pferraro) [reply clustering/infinispan/client/spi/src/main/java/org/wildfly/clustering/infinispan/client/near/CaffeineNearCache.java]: Fixed.  Good catch @jamezp .

**Review by rhusar — APPROVED** [review]
LGTM now

**Review by jamezp — APPROVED** [review]

**Pull request merged**